src/main/scala/treecanopy/Main.scala

```diff
@@ -3,70 +3,127 @@ package treecanopy
 
-    try {
+    val sourceBucket = "azavea-datahub"
+    val statewideLayerName = "raw/treecanopy-2006-2008-pennsylvania-albers"
+    val localLayerName = "raw/treecanopy-pennsylvania-local/harrisburg"
+    val localLayerNames = List(
+      "raw/treecanopy-pennsylvania-local/abingtons",
+      "raw/treecanopy-pennsylvania-local/alleghenyCounty",
+      "raw/treecanopy-pennsylvania-local/harrisburg",
+      "raw/treecanopy-pennsylvania-local/lancasterCity",
+      "raw/treecanopy-pennsylvania-local/montgomeryCounty",
+      "raw/treecanopy-pennsylvania-local/stateCollege"
+    )
+    val outputPath = "s3://com.azavea.datahub.tms/{name}/{z}/{x}/{y}.png"
+    val outputName = "treecanopy-2006-2008-pa-merged"
+
+    val tileSize = 256
+    val targetLayoutScheme =
+      ZoomedLayoutScheme(WebMercator, tileSize)
+
+    // Map 1 to the canopy color and everything else to 0
+    val colorMap =
+      ColorMap(
+        Map(
+          0 -> 0x00000000,
+          1 -> 0x139A68FF,
+          Int.MaxValue -> 0x00000000
+        )
+      )
+
+    val clipTo = Polygon(
+      Point(1257023, 2337147),
+      Point(1257023, 1954954),
+      Point(1793996, 1954954),
+      Point(1793996, 2337147),
+      Point(1257023, 2337147)
+    )
 
-      println(args.toSeq)
-      val etl = Etl(args)
+    val maxLocalZoom = 19  // highest zoom level of local datasets
+    val partitioner = new HashPartitioner(5000)
 
-      val sourceTiles =
-        etl.load[ProjectedExtent, Tile]
+    try {
+      val raw = loadFromS3(sourceBucket, statewideLayerName, partitioner.numPartitions)
+        .filter { case (key, value) => key.extent.intersects(clipTo) }
+      val (myMaxZoom, tiled) = tile(raw, tileSize, NearestNeighbor, partitioner)
+      val resampled = if (myMaxZoom >= maxLocalZoom) {
+        tiled
+      } else {
+        tiled.resampleToZoom(myMaxZoom, maxLocalZoom, NearestNeighbor)
+      }
+      val statewideLevelStream = Pyramid.levelStream(resampled, targetLayoutScheme, maxLocalZoom, NearestNeighbor)
 
-      // Reproject and tile the layer.
-      val (zoom, canopy) =
-        etl.tile(sourceTiles)
+      val raw2 = loadFromS3(sourceBucket, localLayerName, partitioner.numPartitions)
+      val (myMaxZoom2, tiled2) = tile(raw2, tileSize, NearestNeighbor, partitioner)
+      val resampled2 = if (myMaxZoom2 >= maxLocalZoom) {
+        tiled2
+      } else {
+        tiled2.resampleToZoom(myMaxZoom2, maxLocalZoom, NearestNeighbor)
+      }
+      val localLevelStream = Pyramid.levelStream(resampled2, targetLayoutScheme, maxLocalZoom, NearestNeighbor)
 
-      val conf = etl.conf
-      val path = "s3://com.azavea.datahub.tms/{name}/{z}/{x}/{y}.png"
+      statewideLevelStream
+        .zip(localLevelStream)
+        .foreach { case ((z, statewideLayer), (_, localLayer)) =>
+          // merge() will add statewide values only where the local value is absent or NODATA
+          val merged = localLayer.merge(statewideLayer)
```

Comment:
In this simplified version I still haven't been able to get around a "Task not serializable" exception on the above `merge` line. The "Serialization stack" says the offending object is `geotrellis.spark.package$`, which contains the `merge` method.

Any ideas?

The code is based on
https://github.com/lossyrob/geotrellis-ned-example/blob/master/src/main/scala/elevation/Main.scala#L215, which presumably doesn't have the same problem though it doesn't do a merge.

```
6/06/27 14:13:31 ERROR ApplicationMaster: User class threw exception: org.apache.spark.SparkException: Task not serializable
org.apache.spark.SparkException: Task not serializable
	at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:304)
	at org.apache.spark.util.ClosureCleaner$.org$apache$spark$util$ClosureCleaner$$clean(ClosureCleaner.scala:294)
	at org.apache.spark.util.ClosureCleaner$.clean(ClosureCleaner.scala:122)
	at org.apache.spark.SparkContext.clean(SparkContext.scala:2055)
	at org.apache.spark.rdd.RDD$$anonfun$flatMap$1.apply(RDD.scala:333)
	at org.apache.spark.rdd.RDD$$anonfun$flatMap$1.apply(RDD.scala:332)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:150)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:111)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:316)
	at org.apache.spark.rdd.RDD.flatMap(RDD.scala:332)
	at geotrellis.spark.merge.RDDLayoutMergeMethods.merge(RddLayoutMergeMethods.scala:26)
	at treecanopy.Ingest$$anonfun$main$1.apply(Main.scala:88)
	at treecanopy.Ingest$$anonfun$main$1.apply(Main.scala:86)
	at scala.collection.immutable.Stream.foreach(Stream.scala:547)
	at treecanopy.Ingest$.main(Main.scala:86)
	at treecanopy.Ingest.main(Main.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.spark.deploy.yarn.ApplicationMaster$$anon$2.run(ApplicationMaster.scala:542)
Caused by: java.io.NotSerializableException: geotrellis.spark.package$
Serialization stack:
	- object not serializable (class: geotrellis.spark.package$, value: geotrellis.spark.package$@297b65ff)
	- field (class: geotrellis.spark.merge.Implicits$withRDDLayoutMergeMethods, name: $outer, type: interface geotrellis.spark.merge.Implicits)
	- object (class geotrellis.spark.merge.Implicits$withRDDLayoutMergeMethods, geotrellis.spark.merge.Implicits$withRDDLayoutMergeMethods@4930cfab)
	- field (class: geotrellis.spark.merge.RDDLayoutMergeMethods$$anonfun$1, name: $outer, type: class geotrellis.spark.merge.RDDLayoutMergeMethods)
	- object (class geotrellis.spark.merge.RDDLayoutMergeMethods$$anonfun$1, <function1>)
	at org.apache.spark.serializer.SerializationDebugger$.improveException(SerializationDebugger.scala:40)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:47)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:101)
	at org.apache.spark.util.ClosureCleaner$.ensureSerializable(ClosureCleaner.scala:301)
	... 20 more
```